TraceIntelPTBundleSaver.cpp: fails to build on error: no viable conversion from 'const ConstString' to 'const Twine'

recently start to fail on linux:
```

/opt/sccache//sccache /build/source/build-llvm/bin/clang++ -DHAVE_ROUND -D_GLIBCXX_USE_CXX11_ABI=1 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source/Plugins/Trace/intel-pt -I/build/source/lldb/source/Plugins/Trace/intel-pt -I/build/source/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/include -I/build/source/llvm/include -I/build/source/clang/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/../clang/include -I/build/source/lldb/source -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source -isystem /usr/include/libxml2 -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wno-pass-failed -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=../../../../ -ffile-prefix-map=/build/source/= -no-canonical-prefixes -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-vla-extension -O2 -DNDEBUG -g1 -std=c++17 -fno-exceptions -funwind-tables -MD -MT tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleSaver.cpp.o -MF tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleSaver.cpp.o.d -o tools/lldb/source/Plugins/Trace/intel-pt/CMakeFiles/lldbPluginTraceIntelPT.dir/TraceIntelPTBundleSaver.cpp.o -c /build/source/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleSaver.cpp

/home/sylvestre/dev/debian/pkg-llvm/llvm-toolchain/branches/llvm-toolchain-snapshot_23~++20260412053502+b444d1deb508/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleSaver.cpp:275:33: error: no viable conversion from 'const ConstString' to 'const Twine'
  275 |     sys::fs::create_directories(path_to_copy_module.GetDirectory());
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sylvestre/dev/debian/pkg-llvm/llvm-toolchain/branches/llvm-toolchain-snapshot_23~++20260412053502+b444d1deb508/llvm/include/llvm/ADT/Twine.h:250:3: note: candidate constructor not viable: no known conversion from 'const ConstString' to 'const Twine &' for 1st argument
  250 |   Twine(const Twine &) = default;
      |   ^     ~~~~~~~~~~~~~
```

--- a/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleSaver.cpp
+++ b/lldb/source/Plugins/Trace/intel-pt/TraceIntelPTBundleSaver.cpp
@@ -272,7 +272,8 @@ BuildModulesSection(Process &process, FileSpec directory) {
     FileSpec path_to_copy_module = directory;
     path_to_copy_module.AppendPathComponent("modules");
     path_to_copy_module.AppendPathComponent(system_path);
-    sys::fs::create_directories(path_to_copy_module.GetDirectory());
+    sys::fs::create_directories(
+        path_to_copy_module.GetDirectory().GetStringRef());
 
     if (std::error_code ec =
             llvm::sys::fs::copy_file(file, path_to_copy_module.GetPath()))

